packages/zowe-explorer/src/trees/dataset/DatasetTemplates.ts

```diff
@@ -0,0 +1,92 @@
+        ZoweLogger.trace(vscode.l10n.t("Resetting data set templates array."));
+        await this.updateDsTemplateSetting();
+    }
+
+    public static async updateDsTemplateSetting(
+        templates: Types.DataSetAllocTemplate[] = [],
+        target: vscode.ConfigurationTarget = vscode.ConfigurationTarget.Global
+    ): Promise<void> {
+        ZoweLogger.trace(vscode.l10n.t("Updating data set templates."));
+        await SettingsConfig.setDirectValue(Constants.SETTINGS_DS_TEMPLATES, templates, target);
+    }
+
+    public static async addDsTemplateSetting(criteria: Types.DataSetAllocTemplate): Promise<void> {
+        if (criteria) {
+            let newTemplateName: string;
+            Object.entries(criteria).forEach(([key]) => {
+                newTemplateName = key;
+            });
+            let userPick;
+            if (vscode.workspace.workspaceFolders?.[0] != null) {
+                userPick = await this.promptForSaveLocation();
+            }
+            let target = vscode.ConfigurationTarget.Global;
+            if (userPick?.label.includes("Workspace")) {
+                target = vscode.ConfigurationTarget.Workspace;
+            }
+            let templateSettings = this.getTemplatesPerLocation(target);
+            // Remove any entries that match
+            templateSettings = templateSettings.filter((template) => {
+                let historyName: string;
+                Object.entries(template).forEach(([key]) => {
+                    historyName = key;
+                });
+                return historyName !== newTemplateName;
+            });
+            // Add value to front of stack
+            templateSettings.unshift(criteria);
+            ZoweLogger.info(vscode.l10n.t("Adding new data set template {0}.", newTemplateName));
+            await this.updateDsTemplateSetting(templateSettings, target);
+        }
+    }
+
+    // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+    private static promptForSaveLocation() {
+        const qpOptions: vscode.QuickPickOptions = {
+            title: vscode.l10n.t("Data set template save location"),
+            placeHolder: vscode.l10n.t("Choose the setting location to save the data set template..."),
+            ignoreFocusOut: true,
+            canPickMany: false,
+        };
+        const qpItems = [];
+        qpItems.push(new FilterItem({ text: vscode.l10n.t("Save as User setting"), show: true }));
+        qpItems.push(new FilterItem({ text: vscode.l10n.t("Save as Workspace setting"), show: true }));
+        return Gui.showQuickPick(qpItems, qpOptions);
+    }
+
+    private static getTemplatesPerLocation(target: vscode.ConfigurationTarget = vscode.ConfigurationTarget.Global): Types.DataSetAllocTemplate[] {
+        const key = Constants.SETTINGS_DS_TEMPLATES;
+        const [first, ...rest] = key.split(".");
+        const config = vscode.workspace.getConfiguration(first).inspect(rest.join("."));
```

Comment:
Can we reuse `getDsTemplates` or `SettingsConfig.getDirectValue` here to reduce code duplication?

Reply:
this line calls inspect vs get like suggested methods for the configuration to allow check for local vs global